### PR TITLE
feat: Generate advance & annual invoices

### PR DESCRIPTION
**Pull request by GlennLz**

### CODE REVIEW

**Review by DriesOlbrechts — CHANGES_REQUESTED** [review]

File: src/queries/invoice.ts [hunk 1]
```diff
@@ -59,6 +59,18 @@ const getAllInvoices = `
...
+    INSERT INTO ${TABLES.INVOICES} (contract_id, supplier_id, type_id, creation_date, due_date, price, tax, period_start, period_end, tariff_id)
+        VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, 1)
+        RETURNING id
+`;
+
+const insertInvoiceStatus = `
+    INSERT INTO ${TABLES.INVOICES_STATUSES} (invoice_id, status_id)
+        VALUES ($1, $2)
+        RETURNING invoice_id
+`;
```
Comment (DrunkenToast) [inline src/queries/invoice.ts:73]: maybe also add a service function to modify the type.

**Review by DrunkenToast — CHANGES_REQUESTED** [review]
Do you plan on making a route for this? /invoices/id/download and a route to "confirm payment" etc?

File: src/app.ts [hunk 2]
```diff
@@ -35,6 +37,8 @@ try {
     process.on('SIGTERM', () => {onClose(server)});
       
     // scheduleInvoiceJobs();
+    //await generateInvoices(INVOICE_TYPE.DEBIT);
+    //await generateInvoices(INVOICE_TYPE.ADVANCE);
```
[the remark this answers is not in the document]
Comment (GlennLz) [reply src/app.ts:41]: It's to run the algorithm without having to wait for the scheduler

File: src/models/invoice.ts [hunk 3]
```diff
@@ -13,8 +13,8 @@ export interface Invoice {
     period_end: Date,
     status: INVOICE_STATUS,
     type: INVOICE_TYPE,
-    address: Address,
-    customer: Customer,
+    address: Address | null,
+    customer: Customer | null,
```
[the remark this answers is not in the document]
Comment (GlennLz) [reply src/models/invoice.ts:17]: done

**Review by DrunkenToast — APPROVED** [review]

**Review by DrunkenToast — APPROVED** [review]

**Review by DriesOlbrechts — APPROVED** [review]

**Pull request merged**